lib/trino-matching/src/main/java/io/trino/matching/CapturePattern.java

```diff
@@ -34,9 +34,9 @@ public Capture<T> capture()
     }
 
     @Override
-    public <C> Stream<Match> accept(Object object, Captures captures, C context)
+    public <C> Iterable<Match> accept(Object object, Captures captures, C context)
```

Comment:
What's the benefit of using Iterable instead of Stream ?
I thought Stream is usually preferred in modern java code